File: my_flutter_app/lib/screens/fiat_currencies_screen.dart
import 'package:flutter/material.dart';
import 'package:shared_preferences/shared_preferences.dart';
import '../layout/bottom_menu_with_siri.dart';

class FiatCurrenciesScreen extends StatelessWidget {
  const FiatCurrenciesScreen({Key? key}) : super(key: key);

  Future<void> _saveSelectedCurrency(BuildContext context, String currency) async {
    final prefs = await SharedPreferences.getInstance();
    await prefs.setString('selected_currency', currency);
  }

  @override
  Widget build(BuildContext context) {
    final List<Map<String, dynamic>> currencies = [
      {
        'code': 'USD',
        'symbol': '\$',
        'flag': 'assets/images/us.png',
        'country': 'United States',
        'fullName': 'US Dollar',
      },
      {
        'code': 'CAD',
        'symbol': 'CA',
        'flag': 'assets/images/ca.png',
        'country': 'Canada',
        'fullName': 'Canadian Dollar',
      },
      {
        'code': 'AUD',
        'symbol': 'AU',
        'flag': 'assets/images/au.png',
        'country': 'Australia',
        'fullName': 'Australian Dollar',
      },
      {
        'code': 'GBP',
        'symbol': '£',
        'flag': 'assets/images/gb.png',
        'country': 'United Kingdom',
        'fullName': 'Pound Sterling',
      },
      {
        'code': 'EUR',
        'symbol': '€',
        'flag': 'assets/images/eu.png',
        'country': 'European Union',
        'fullName': 'Euro',
      },
      {
        'code': 'KWD',
        'symbol': 'KD',
        'flag': 'assets/images/kw.png',
        'country': 'Kuwait',
        'fullName': 'Kuwaiti Dinar',
      },
      {
        'code': 'TRY',
        'symbol': '₺',
        'flag': 'assets/images/tr.png',
        'country': 'Turkey',
        'fullName': 'Turkish Lira',
      },
      {
        'code': 'SAR',
        'symbol': '﷼',
        'flag': 'assets/images/sa.png',
        'country': 'Saudi Arabia',
        'fullName': 'Saudi Riyal',
      },
      {
        'code': 'CNY',
        'symbol': '¥',
        'flag': 'assets/images/cn.png',
        'country': 'China',
        'fullName': 'Chinese Yuan',
      },
      {
        'code': 'KRW',
        'symbol': '₩',
        'flag': 'assets/images/kr.png',
        'country': 'South Korea',
        'fullName': 'South Korean Won',
      },
      {
        'code': 'JPY',
        'symbol': '¥',
        'flag': 'assets/images/jp.png',
        'country': 'Japan',
        'fullName': 'Japanese Yen',
      },
      {
        'code': 'INR',
        'symbol': '₹',
        'flag': 'assets/images/in.png',
        'country': 'India',
        'fullName': 'Indian Rupee',
      },
      {
        'code': 'RUB',
        'symbol': '₽',
        'flag': 'assets/images/ru.png',
        'country': 'Russia',
        'fullName': 'Russian Ruble',
      },
      {
        'code': 'IQD',
        'symbol': 'ع.د',
        'flag': 'assets/images/iq.png',
        'country': 'Iraq',
        'fullName': 'Iraqi Dinar',
      },
      {
        'code': 'TND',
        'symbol': 'د.ت',
        'flag': 'assets/images/tn.png',
        'country': 'Tunisia',
        'fullName': 'Tunisian Dinar',
      },
      {
        'code': 'BHD',
        'symbol': 'ب.د',
        'flag': 'assets/images/bh.png',
        'country': 'Bahrain',
        'fullName': 'Bahraini Dinar',
      },
    ];

    return Scaffold(
      backgroundColor: Colors.white,
      appBar: AppBar(
        backgroundColor: Colors.white,
        elevation: 0,
        title: const Text(
          'Fiat Currencies',
          style: TextStyle(
            color: Colors.black,
            fontWeight: FontWeight.bold,
            fontSize: 24,
          ),
        ),
        iconTheme: const IconThemeData(color: Colors.black),
      ),
      body: Padding(
        padding: const EdgeInsets.symmetric(horizontal: 16.0),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            const SizedBox(height: 24),
            const Text(
              'All currency:',
              style: TextStyle(
                fontSize: 16,
                color: Color(0xCB838383),
                fontWeight: FontWeight.w600,
              ),
            ),
            const SizedBox(height: 24),
            Expanded(
              child: ListView.builder(
                itemCount: currencies.length,
                itemBuilder: (context, index) {
                  final currency = currencies[index];
                  return InkWell(
                    onTap: () async {
                      await _saveSelectedCurrency(context, currency['code']);
                      Navigator.pop(context);
                    },
                    child: Padding(
                      padding: const EdgeInsets.symmetric(vertical: 10.0),
                      child: Row(
                        crossAxisAlignment: CrossAxisAlignment.start,
                        children: [
                          Image.asset(
                            currency['flag'],
                            width: 28,
                            height: 28,
                          ),
                          const SizedBox(width: 12),
                          Expanded(
                            child: Column(
                              crossAxisAlignment: CrossAxisAlignment.start,
                              children: [
                                Row(
                                  children: [
                                    Text(
                                      '${currency['code']} (${currency['symbol']})',
                                      style: const TextStyle(
                                        fontSize: 18,
                                        color: Colors.black,
                                        fontWeight: FontWeight.w500,
                                      ),
                                    ),
                                    const Spacer(),
                                    Text(
                                      currency['fullName'],
                                      style: const TextStyle(
                                        fontSize: 15,
                                        color: Colors.black,
                                        fontWeight: FontWeight.w400,
                                      ),
                                    ),
                                  ],
                                ),
                                const SizedBox(height: 2),
                                Text(
                                  currency['country'],
                                  style: const TextStyle(
                                    fontSize: 13,
                                    color: Colors.grey,
                                    fontWeight: FontWeight.w400,
                                  ),
                                ),
                              ],
                            ),
                          ),
                        ],
                      ),
                    ),
                  );
                },
              ),
            ),
          ],
        ),
      ),
      bottomNavigationBar: const BottomMenuWithSiri(),
    );
  }
} 

File: ۱/FiatCurrenciesScreen.kt
package com.laxce.adl.ui.theme.screen

import androidx.compose.foundation.Image
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.navigation.NavController
import com.laxce.adl.R
import com.laxce.adl.ui.theme.layout.MainLayout
import com.laxce.adl.utility.saveSelectedCurrency

@Composable
fun FiatCurrenciesScreen(navController: NavController) {

    MainLayout(navController = navController) {
        Column(modifier = Modifier.fillMaxSize().background(Color.White)) {
            Column(
                modifier = Modifier
                    .fillMaxSize()
                    .padding(16.dp)
            ) {
                // Header
                Text(
                    text = "Fiat Currencies",
                    fontSize = 24.sp,
                    fontWeight = FontWeight.Bold,
                    modifier = Modifier.padding(bottom = 16.dp)
                )

                // Box containing the currencies list
                Box(
                    modifier = Modifier
                        .fillMaxSize()
                        .padding(8.dp)
                ) {
                    Column {
                        Text(
                            text = "All currency:",
                            fontSize = 16.sp,
                            color = Color(0xCB838383),
                            fontWeight = FontWeight.SemiBold,
                            modifier = Modifier.padding(bottom = 16.dp)
                        )

                        LazyColumn(modifier = Modifier.fillMaxSize()) {
                            val currencies = listOf(
                                Triple("USD", "$" , R.drawable.us),
                                Triple("CAD", "CA", R.drawable.ca),
                                Triple("AUD", "AU", R.drawable.au),
                                Triple("GBP", "£", R.drawable.gb),
                                Triple("EUR", "€", R.drawable.eu),
                                Triple("KWD", "KD", R.drawable.kw),
                                Triple("TRY", "₺", R.drawable.tr),
//                                Triple("IRR", "﷼", R.drawable.ir),
                                Triple("SAR", "﷼", R.drawable.sa),
                                Triple("CNY", "¥", R.drawable.cn),
                                Triple("KRW", "₩", R.drawable.kr),
                                Triple("JPY", "¥", R.drawable.jp),
                                Triple("INR", "₹", R.drawable.`in`),
                                Triple("RUB", "₽", R.drawable.ru),
                                Triple("IQD", "ع.د", R.drawable.iq),
                                Triple("TND", "د.ت", R.drawable.tn),
                                Triple("BHD", "ب.د", R.drawable.bh)
                            )

                            items(currencies.size) { index ->
                                val (currency, symbol, flagRes) = currencies[index]
                                Row(
                                    modifier = Modifier
                                        .fillMaxWidth()
                                        .clickable {
                                            saveSelectedCurrency(
                                                context = navController.context,
                                                currency = currency
                                            )
                                            navController.popBackStack()
                                        }
                                        .padding(vertical = 8.dp),
                                    horizontalArrangement = Arrangement.Start
                                ) {
                                    // Flag image
                                    Image(
                                        painter = painterResource(id = flagRes),
                                        contentDescription = "$currency flag",
                                        modifier = Modifier.size(24.dp)
                                    )

                                    Spacer(modifier = Modifier.width(8.dp))

                                    // Currency details
                                    Column {
                                        Text(
                                            text = "$currency ($symbol)",
                                            fontSize = 18.sp,
                                            color = Color.Black
                                        )
                                    }
                                }
                            }
                        }
                    }
                }
            }
        }
    }
}
